sdk/src/mcm_dp.c

```diff
@@ -6,17 +6,120 @@
 #include <bsd/string.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <errno.h>
 
 #include "memif_impl.h"
 #include "udp_impl.h"
 #include "logger.h"
 #include "mcm_dp.h"
 #include "media_proxy_ctrl.h"
 
+/* Calculate audio buffer size based on ST2110-30 related parameters */
+static int mcm_calc_audio_buffer_size(mcm_audio_args *params, uint32_t *size)
+{
+    uint32_t sample_size = 0;
+    uint32_t sample_num = 0;
+
+    switch (params->format) {
+    case AUDIO_FMT_PCM8:
+        sample_size = 1;
+        break;
+    case AUDIO_FMT_PCM16:
+        sample_size = 2;
+        break;
+    case AUDIO_FMT_PCM24:
+        sample_size = 3;
+        break;
+    case AUDIO_FMT_AM824:
+        sample_size = 4;
+        break;
+    default:
+        *size = 0;
+        return -EINVAL;
+    }
+
+    switch (params->sampling) {
```

Comment:
Same here. Is there any way to automate this formula somehow? Dependency between sampling and sample_num value?